lib/filterx/tests/test_expr_condition.c

```diff
@@ -0,0 +1,470 @@
+/*
```

Comment:
There are a lot of memleaks, some of them originate from the unit test, but maybe some come from code problems.

Please run the tests with valgrind and fix the memleaks.
```
valgrind --leak-check=full --trace-children=yes lib/filterx/tests/test_expr_condition
```